lib/template.html

```diff
@@ -5,21 +5,13 @@
 {% endif %}
 
 {% assign seo_site_title = site.title | default: site.name %}
+{% assign seo_page_title = page.title | default: seo_site_title %}
+{% assign seo_title = page.title | default: seo_site_title %}
 
-{% if page.title %}
-  {% assign seo_title = page.title %}
-  {% assign seo_page_title = page.title %}
-
-  {% if seo_site_title %}
-    {% assign seo_title = seo_title | append:" - " | append: seo_site_title %}
-  {% endif %}
-{% elsif seo_site_title %}
-  {% assign seo_title = seo_site_title %}
-  {% assign seo_page_title = seo_site_title %}
-
-  {% if site.description %}
-    {% assign seo_title = seo_title | append:" - " | append: site.description %}
-  {% endif %}
+{% if page.title and seo_site_title %}
+  {% assign seo_title = seo_title | append:" | " | append: seo_site_title %}
+{% elsif site.description and seo_site_title %}
+  {% assign seo_title = seo_title | append:" | " | append: site.description %}
```

Comment:
Agreed. Updated with https://github.com/jekyll/jekyll-seo-tag/pull/162/commits/2a48f272f68b309b108b4cda27a862e6744c1f20. Had to keep L9 as it contains some default logic which, if would remove, would break some of the tests introduced.